src/Compilers/Core/Portable/SourceGeneration/Nodes/IIncrementalGeneratorOutputNode.cs

```diff
@@ -19,9 +19,18 @@ internal interface IIncrementalGeneratorOutputNode
         void AppendOutputs(IncrementalExecutionContext context, CancellationToken cancellationToken);
     }
 
-    internal enum IncrementalGeneratorOutputKind
+    /// <summary>
+    /// Represents the various output kinds of an <see cref="IIncrementalGenerator"/>. 
+    /// </summary>
+    /// <remarks>
+    /// Can be passed as a bit field when creating a <see cref="GeneratorDriver"/> to selectively disable outputs.
+    /// </remarks>
+    [Flags]
+    public enum IncrementalGeneratorOutputKind
     {
-        Source,
-        PostInit
+        None = 0,
+        Source = 0b1,
+        PostInit = 0b10,
+        Implementation = 0b100
```

Comment:
Need to add a doc comment to this to describe what they do given that they are public values.